snuba/subscriptions/verifier.py

```diff
@@ -99,3 +109,202 @@ def close(self, timeout: Optional[float] = None) -> None:
+
+        self.__orig_consumer_last_message: Optional[
+            Message[SubscriptionResultData]
+        ] = None
+        self.__new_consumer_last_message: Optional[
+            Message[SubscriptionResultData]
+        ] = None
+
+    def _build_result_consumer(
+        self, consumer_group: str, auto_offset_reset: str
+    ) -> Consumer[SubscriptionResultData]:
+        return SubscriptionResultConsumer(
+            KafkaConsumer(
+                build_kafka_consumer_configuration(
+                    None, group_id=consumer_group, auto_offset_reset=auto_offset_reset,
+                ),
+            )
+        )
+
+    def subscribe(
+        self,
+        topics: Sequence[Topic],
+        on_assign: Optional[Callable[[Mapping[Partition, int]], None]] = None,
+        on_revoke: Optional[Callable[[Sequence[Partition]], None]] = None,
+    ) -> None:
+        """
+        Note the topic actually passed here is ignored as each consumer subscribes to a different topic
+        """
+        self.__orig_topic_consumer.subscribe(
+            [self.__orig_topic], on_assign=on_assign, on_revoke=on_revoke
+        )
+        self.__new_topic_consumer.subscribe(
+            [self.__new_topic], on_assign=on_assign, on_revoke=on_revoke
+        )
+
+    def unsubscribe(self) -> None:
+        self.__orig_topic_consumer.unsubscribe()
+        self.__new_topic_consumer.unsubscribe()
+
+    def poll(
+        self, timeout: Optional[float] = None
+    ) -> Optional[Message[SubscriptionResultData]]:
+        if self.__orig_consumer_last_message is None:
+            self.__orig_consumer_last_message = self.__orig_topic_consumer.poll(timeout)
+
+        if self.__new_consumer_last_message is None:
+            self.__new_consumer_last_message = self.__new_topic_consumer.poll(timeout)
+
+        message: Optional[Message[SubscriptionResultData]] = None
+
+        if (
+            self.__orig_consumer_last_message is None
+            and self.__new_consumer_last_message is None
+        ):
+            pass
+        elif self.__new_consumer_last_message is None:
+            message = self.__orig_consumer_last_message
+            self.__orig_consumer_last_message = None
+        elif self.__orig_consumer_last_message is None:
+            message = self.__new_consumer_last_message
+            self.__new_consumer_last_message = None
+        else:
+            # Return the earlier message based on scheduled timestamp
+            # (or the new topic message if both are the same)
+            orig_payload = self.__orig_consumer_last_message.payload
+            new_payload = self.__new_consumer_last_message.payload
+
+            if orig_payload.timestamp < new_payload.timestamp:
+                message = self.__orig_consumer_last_message
+                self.__orig_consumer_last_message = None
+            else:
+                message = self.__new_consumer_last_message
+                self.__new_consumer_last_message = None
```

Comment:
This consumer doesn't really provide any guarantees, the idea was just to make the two topics move at very approximately the same speed. On second thoughts we can probably just achieve this with a single consumer though since there's not a lot that's being guaranteed by the synchronized consumer.